src/mongocxx/exception/operation_exception.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <mongocxx/exception/exception.hpp>
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+class MONGOCXX_API operation_exception : public exception {
+   public:
+   	using exception::exception;
+
+    ///
+    /// Constructs a new operation exception.
+    ///
+    /// @param ec
+    ///   The error code associated with this exception.
+    /// @param raw_server_error
+    ///   The optional raw bson error document to be associated with this exception.
+    /// @param what_arg
+    ///   An optional message to be returned by `what`.
+    ///
+    explicit operation_exception(std::error_code ec, bsoncxx::document::value&& raw_server_error,
```

Comment:
I don't think this needs to be explicit since there are multiple non-default arguments